apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/LoadAppOptions.java

```diff
@@ -154,17 +85,60 @@ public String toString() {
     builder.append(idHasherConfig.getHashIterations());
     builder.append(", hicnHashPepper=");
     builder.append("***");
-    builder.append(", loaderThreads=");
-    builder.append(loaderThreads);
     builder.append(", idempotencyRequired=");
     builder.append(idempotencyRequired);
     builder.append(", filteringNonNullAndNon2023Benes=");
     builder.append(filteringNonNullAndNon2023Benes);
-    builder.append(", recordBatchSize=");
-    builder.append(recordBatchSize);
-    builder.append(", taskQueueSizeMultiple=");
-    builder.append(taskQueueSizeMultiple);
+    builder.append(", beneficiaryPerformanceSettings=");
+    builder.append(beneficiaryPerformanceSettings);
+    builder.append(", claimPerformanceSettings=");
+    builder.append(claimPerformanceSettings);
     builder.append("]");
     return builder.toString();
   }
+
+  /**
+   * Select the appropriate {@link PerformanceSettings} to use for the given {@link RifFileType}.
+   *
+   * @param fileType type of rif data being loaded
+   * @return appropriate performance settings
+   */
+  public PerformanceSettings selectPerformanceSettingsForFileType(RifFileType fileType) {
+    return switch (fileType) {
+      case BENEFICIARY, BENEFICIARY_HISTORY -> beneficiaryPerformanceSettings;
+      default -> claimPerformanceSettings;
+    };
+  }
+
+  /** Settings used for performance tuning of the {@link RifLoader}. */
+  @Data
+  public static class PerformanceSettings implements Serializable {
+    /** The number of loader threads. */
+    private final int loaderThreads;
+
+    /**
+     * The number of {@link RifRecordEvent}s that will be included in each processing batch. Note
+     * that larger batch sizes mean that more {@link RifRecordEvent}s will be held in memory
+     * simultaneously.
+     */
+    private final int recordBatchSize;
+
+    /** The maximum size (per thread) of the task queue used to process batches. */
+    private final int taskQueueSizeMultiple;
+
+    /**
+     * Initializes an instance.
+     *
+     * @param loaderThreads the value to use for {@link #loaderThreads}
+     * @param recordBatchSize the load batch size
+     * @param taskQueueSizeMultiple the task queue size multiple
+     */
+    public PerformanceSettings(int loaderThreads, int recordBatchSize, int taskQueueSizeMultiple) {
+      if (loaderThreads < 1) throw new IllegalArgumentException();
+      if (taskQueueSizeMultiple < 1) throw new IllegalArgumentException();
```

Comment:
I think there weren't tests for these failure cases.

Reply:
Decided this logic was best handled in `AppConfiguration`.  Moved it there and added tests.  Good showcase for using assertj as well.